File: modules/spring-core/testdata/inspection/autowired/src/component_scan/many_beans/TestAutowired.java
package component_scan.many_beans;

import org.springframework.context.annotation.Bean;
import org.springframework.stereotype.Component;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.Primary;

@Configuration
class MyComponent {
    @Autowired
    FooInterface one;

    @Autowired
    FooInterface nameComponent;

    @Autowired
    BarInterface primer;
}

@Component
class FooBeanComponent {
    @Autowired FooBean1 fooBean1;
    @Autowired FooBeanInterface fooBeanInterface;
    @Autowired FooBean2 fooBean2;
}
interface FooInterface {}

@Component
class FooBean implements FooInterface {}
@Component
class OtherBean implements FooInterface {}
@Component("nameComponent")
class ThreeBean implements FooInterface {}

interface BarInterface {}
@Component
class BarImpl implements BarInterface {}
@Component
@Primary
class PrimerBarImpl implements BarInterface {}

@Configuration
class TestConfiguration {
    @Bean FooBean1 createBean1() {
        return new FooBean1();
    }

    @Primary
    @Bean FooBean1 fooBeanPrimary() {
        return new FooBean1();
    }

    @Bean FooBean2 createBean2() {
        return new FooBean2();
    }
}
interface FooBeanInterface {}
class FooBean1 implements FooBeanInterface {}
class FooBean2 implements FooBeanInterface {}

File: modules/spring-core/testdata/kotlin/inspection/autowired/src/component_scan/many_beans/TestAutowired.kt
package component_scan.many_beans

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Primary
import org.springframework.stereotype.Component

@Configuration
open class MyComponent {
    @Autowired
    var one: FooInterface? = null

    @Autowired
    var nameComponent: FooInterface? = null

    @Autowired
    var primer: BarInterface? = null
}

@Component
open class FooBeanComponent {
    @Autowired
    var fooBean1: FooBean1? = null

    @Autowired
    var fooBeanInterface: FooBeanInterface? = null

    @Autowired
    var fooBean2: FooBean2? = null
}

interface FooInterface

@Component
open class FooBean : FooInterface

@Component
open class OtherBean : FooInterface

@Component("nameComponent")
internal class ThreeBean : FooInterface

internal interface BarInterface

@Component
internal class BarImpl : BarInterface

@Component
@Primary
internal class PrimerBarImpl : BarInterface

@Configuration
open class TestConfiguration {
    @Bean
    open fun createBean1(): FooBean1 {
        return FooBean1()
    }

    @Primary
    @Bean
    open fun fooBeanPrimary(): FooBean1 {
        return FooBean1()
    }

    @Bean
    open fun createBean2(): FooBean2 {
        return FooBean2()
    }
}

interface FooBeanInterface
class FooBean1 : FooBeanInterface
class FooBean2 : FooBeanInterface